### PR TITLE
Revert "HW 41 Polina"

### PR DESCRIPTION
**Pull request by jathpr**
Reverts jathpr/FE42Bloc3#63

### CODE REVIEW

**Review by jathpr — COMMENTED** [review]
you named commit `Changes`
seriously?)
https://xkcd.com/1296/

File: my-app/src/App.tsx [hunk 1]
```diff
@@ -13,61 +16,53 @@ type User = {
...
 }
 
-type Pages = 'Auth' | 'Reg' | 'Posts'
-
 export const App = () => {
   const [users, setUsers] = useState<User[]>([])
-  const [page, setPage] = useState<Pages>('Reg')
+  const [searchResult, setSearchResult] = useState('')
+
+  const navigate = useNavigate()
```
Comment (jathpr) [inline my-app/src/App.tsx]: try to separate different logic into different components
here you have auth, navigation, and search

File: my-app/src/App.tsx [hunk 2]
```diff
@@ -13,61 +16,53 @@ type User = {
...
-            {page === 'Reg' && <Registration onReg={addUser} />}
-            {page === 'Auth' && <Authorization onAuth={checkUser} onSignUp={navToReg} />}
+            <Routes>
+              <Route path='Reg' element={<Registration onReg={addUser} />} />
+              <Route path='Auth' element={<Authorization onAuth={checkUser} />} />
+            </Routes>
           </div>
           <div>
-            {page === 'Posts' && <PostsList />}
+            <Routes>
```
Comment (jathpr) [inline my-app/src/App.tsx:60]: why do you place it in a different `Routes`?

File: my-app/src/Components/Post/PostsList.tsx [hunk 3]
```diff
@@ -1,14 +1,20 @@
...
-    useEffect(() => {getPosts({'limit': 4, 'search': 'ast'}).then((resp) => setCards(resp))}, [])
+    useEffect(() => {
+        getPosts({ 'limit': 4, 'search': searchResult }).then((resp) => setCards(resp))
+    }, [searchResult])
+
+    const navigate = useNavigate()
 
     return <>
-    {cards.map(post => <RenderPost post={post} key={post.id} />)}
+        {cards.map(post => <li onClick={() => navigate('/Posts/' + post.id)}><RenderPost post={post} key={post.id} /></li>)}
```
Comment (jathpr) [inline my-app/src/Components/Post/PostsList.tsx:18]: here Link will be a better solution

File: my-app/src/Components/Post/RenderPost.tsx [hunk 4]
```diff
@@ -1,16 +1,28 @@
...
+import { OnePost } from './getPosts'
 import './Posts.css'
+import { useEffect, useState } from 'react'
+import { getPost } from './getPosts'
 
-export const RenderPost = (props: { post: Post }) => (
-    <div className="mainDiv">
+export const RenderPost = (props: { post?: OnePost }) => {
+    const { postId } = useParams()
+    const [onePost, setPost] = useState(props.post)
```
Comment (jathpr) [inline my-app/src/Components/Post/RenderPost.tsx:10]: setPost -> setOnePost

File: my-app/src/Components/Search/Search.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+export const Search = ({onSearch}: Props) => {
+    const [searchResult, setSearchResult] = useState('')
+
+    return (
+        <>
+            <input className='searchInput' type='search' placeholder="Search..." value= {searchResult} onChange={(e) => { setSearchResult(e.currentTarget.value)}}></input>
+            <button className='searchButton' onClick={() => onSearch(searchResult)}>
+                <svg width="24" height="24" viewBox="0 0 21 21" xmlns="http://www.w3.org/2000/svg">
```
Comment (jathpr) [inline my-app/src/Components/Search/Search.tsx]: prefer to move `svg` to separate file

**Review by jathpr — COMMENTED** [review]
zaebishche

**Review by polishukova — APPROVED** [review]

File: my-app/src/App.tsx [hunk 6]
```diff
@@ -13,61 +16,53 @@ type User = {
...
   const addUser = (login: string, password: string) => {
     setUsers([...users, { login, password }])
-    // setPage('Auth')
   }
 
   const checkUser = (login: string, password: string) => {
     const findUser = users.find(user => user.login === login && user.password === password)
-    if (findUser){
-      setPage('Posts')
+    if (findUser) {
```
[the remark this answers is not in the document]
Comment (polishukova) [reply my-app/src/App.tsx:31]: Done

File: my-app/src/Components/Post/PostsList.tsx (hunk 3 above)
Comment (polishukova) [reply my-app/src/Components/Post/PostsList.tsx:18]: Yeap

File: my-app/src/Components/Search/Search.tsx (hunk 5 above)
Comment (polishukova) [reply my-app/src/Components/Search/Search.tsx]: I'll try )

**Pull request merged**